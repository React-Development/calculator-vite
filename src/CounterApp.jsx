import PropTypes from "prop-types";
import { useState } from "react";

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleOperation = (e, operation) => {
    switch (operation) {
      case "add":
        return setCounter((c) => c + 1);
      case "decrease":
        return setCounter((c) => c - 1);
      case "reset":
        return setCounter(value);
      default:
        return;
    }
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2> {counter} </h2>

      <button onClick={(e) => handleOperation(e, "add")}>+1</button>
      <button onClick={(e) => handleOperation(e, "decrease")}>-1</button>
      <button onClick={(e) => handleOperation(e, "reset")}>Reset</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
