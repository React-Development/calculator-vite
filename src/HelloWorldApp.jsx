import PropTypes from "prop-types";

export const HelloWorldApp = ({ title, subtitle, name }) => {
  return (
    <>
      <h1>{title}</h1>
      <p>{subtitle}</p>
      <p>{name}</p>
    </>
  );
};

HelloWorldApp.propTypes = {
  title: PropTypes.string,
  subtitle: PropTypes.string,
};

HelloWorldApp.defaultProps = {
  title: "There is no title.",
  subtitle: "There is no subtitle",
  name: "Brayan Garcia",
};
