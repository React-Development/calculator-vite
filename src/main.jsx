import React from "react";
import ReactDOM from "react-dom/client";
import "./styles.css";

import { HelloWorldApp } from "./HelloWorldApp";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <HelloWorldApp title="Hola, soy Goku" />
  </React.StrictMode>
);
