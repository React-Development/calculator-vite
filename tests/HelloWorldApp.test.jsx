import React from "react";
import { render } from "@testing-library/react";

import { HelloWorldApp } from "../src/HelloWorldApp";

describe("tests on HelloWorldApp", () => {
  test("must match with the snapshot", () => {
    const title = "Hello, I'm Goku";
    const { container } = render(<HelloWorldApp title={title} />);

    expect(container).toMatchSnapshot();
  });

  test("should show title on h1", () => {
    const title = "Hello, I'm Goku";
    const { container, getByText } = render(<HelloWorldApp title={title} />);

    expect(getByText(title)).toBeTruthy();

    const h1 = container.querySelector("h1");
    expect(h1.innerHTML).toContain(title);
  });
});
