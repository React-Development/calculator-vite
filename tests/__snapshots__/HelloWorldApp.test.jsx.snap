// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on HelloWorldApp must match with the snapshot 1`] = `
<div>
  <div
    data-testid="test-title"
  >
     
    Hello, I'm Goku
     
  </div>
  <p>
    There is no subtitle
  </p>
  <p>
    There is no subtitle
  </p>
  <p>
    Brayan Garcia
  </p>
</div>
`;
